Adds support for annotation parsing

artman opened the pull request:
- Annotations are supported on: Typdefs, Consts, Enums, Enum values, Structs (& Unions, Exceptions), Struct Fields, Services and Functions
- Updates test to stringify and subsequently parse the Thrift doc in order to get rid of undefined values
- Adds extensive unit test for annotations
- Adds correct error messages when annotations aren't syntactically correct

This PR opts to diverge from the coding style of the rest of the library regarding annotations. Annotations are declared as optional parameters and default to undefined instead of null. This has the following upsides:

- Undefined values aren't omitted when stringifying to JSON
- Existing unit tests continue to pass

Commit:
Adds support for annotation parsing

- Updates the parser to support annotations and string constants
- Updates field parsing to support annotations
- Adds test case for annotation parsing

kevin-greene-ck commented:
@artman It's looking good. Thanks for the effort.

kevin-greene-ck commented on src/main/parser.ts:
```diff
@@ -389,6 +392,32 @@ export function createParser(tokens: Array<Token>, report: ErrorReporter = noopR
+          const nameToken: Token = consume(SyntaxType.Identifier)
+          consume(SyntaxType.EqualToken)
+          const next: Token = advance()
+          const value: StringConstant = parseStringValue(next)
```
I am curious what we gain from adding StringConstant instead of using a StringLiteral as the value of an Annotation.

artman replied:
You're right, there's no need to add a new StringConstant type. Wasn't very familiar with the code-base, so thought we'd want to return constant types like Floats and Doubles do, but given that e.g. include definitions just use StringLiterals, I'll remove the StringConstant.

Commit:
Added annotation support for all applicable Thrift types

This diff opts to diverge from the coding style of the rest of the library regarding annotations. Annotations are declared as optional parameters and default to undefined instead of null. This has the following upsides:

- Undefined values aren't omitted when stringifying to JSON
- Existing unit tests continue to pass

This diff adds the following functionality

- Annotations are supported on: Typdefs, Consts, Enums, Enum values, Structs (& Unions, Exceptions), Struct Fields, Services and Functions
- Updates test to stringify and subsequently parse the Thrift doc in order to get rid of undefined values
- Adds extensive unit test for annotations
- Adds correct error messages when annotations aren't syntactically correct

Commit:
Fixed indentation issues with types.ts

kevin-greene-ck commented:
Excellent work, I'll get this out in a release today.

artman commented:
Great, thanks!
